### ISSUE TITLE
with has undocumented behaviour

### ISSUE THREAD

**Issue by ajb-in** [lstrojny/functional-php#95]
This will not work as documented:

```php
$f = static function ($x) { return $x + 1; };
with($f, partial_any('map', [1, 2, 3], …));
```

**Comment by lstrojny** [comment]
@ajb-in thank you, good point. Have a look at the commit, what do you think?

**Comment by ajb-in** [comment]
`with` should behave like `with_value`, period. If the value is callable, calling it should be the job of `with`'s second argument.
Instead of `with_func($f, $g)` you could have `with($f, partial_any(compose($g, …)))`.
Of course that breaks BC, but it could be a thing for the next major. No strong opinion on how to handle the transition.

**Comment by ajb-in** [comment]
Of course I'm an idiot; 496f677 does _just that_. LGTM.

**Comment by lstrojny** [comment]
Fixed in 1.2.1
